### PR TITLE
Win conditions spike

### PR DESCRIPTION
**Pull request by wallacebilly1**

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -2,15 +2,16 @@ class Board
...
-    @visual = {
-      "A" => %w[. . . . . .],
-      "B" => %w[. . . . . .],
-      "C" => %w[. . . . . .],
-      "D" => %w[. . . . . .],
-      "E" => %w[. . . . . .],
-      "F" => %w[. . . . . .],
-      "G" => %w[. . . . . .]
-    }
+    @visual = Hash.new { |hash, key| hash[key] = Array.new(6, '.') }
```
Comment (grantdavis303) [inline lib/board.rb:5]: Love this one line to create the hash of arrays - way simpler!

File: lib/board.rb [hunk 2]
```diff
@@ -2,15 +2,16 @@ class Board
...
+    # @visual = {
+    #   "A" => %w[. . . . . .],
+    #   "B" => %w[. . . . . .],
+    #   "C" => %w[. . . . . .],
+    #   "D" => %w[. . . . . .],
+    #   "E" => %w[. . . . . .],
+    #   "F" => %w[. . . . . .],
+    #   "G" => %w[. . . . . .]
+    # }
```
Comment (JaredMHobson) [inline lib/board.rb:14]: Like the cool new way to generate the board. Will need to clean up commented out old board

File: lib/game.rb [hunk 3]
```diff
@@ -10,23 +10,24 @@ def initialize
...
     answer = gets.chomp.downcase
-        if answer == 'q'
-            play = false
-        elsif answer == 'p'
-            play
-        else
-          puts @message_bot.invalid_command
-          answer = gets.chomp.downcase
-        end    
+    until play
```
Comment (grantdavis303) [inline lib/game.rb:15]: This is an awesome until loop :)

File: lib/game.rb [hunk 4]
```diff
@@ -10,23 +10,24 @@ def initialize
...
+      if answer == 'q'
+        exit
+      elsif answer == 'p'
+        play = true
+      else
+        puts @message_bot.invalid_command
+        answer = gets.chomp.downcase
+      end
+    end
```
Comment (JaredMHobson) [inline lib/game.rb:24]: Nice job getting this to work

File: lib/game.rb [hunk 5]
```diff
@@ -85,14 +86,34 @@ def is_column_full?(input)
...
+      sleep(5)
+      @board = Board.new
+      @turns_taken = 0
+      start
+    elsif win?
+      puts @message_bot.win(current_player)
+      sleep(5)
+      @board = Board.new
+      @turns_taken = 0
+      start
```
Comment (JaredMHobson) [inline lib/game.rb:101]: Split board reset code into its own method

File: lib/game.rb [hunk 6]
```diff
@@ -85,14 +86,34 @@ def is_column_full?(input)
     @board.visual[input].none?(".")
   end
 
-  def win?
-    if sw_to_ne_diagonal_win? || se_to_nw_diagonal_win? || horizontal_win? || vertical_win?
-      'True'
-    else
-      'False'
+  def game_over
```
Comment (grantdavis303) [inline lib/game.rb:89]: We'll add some sweet ASCII art when the winner is declared, but nice job on getting the game_over method to check if a win condition was met OR if the game resulted in a draw

File: lib/game.rb [hunk 7]
```diff
@@ -85,14 +86,34 @@ def is_column_full?(input)
...
+      sleep(5)
+      @board = Board.new
+      @turns_taken = 0
+      start
     end
   end
 
+  def check_for_draw?
+    @turns_taken > 42
+  end
```
Comment (JaredMHobson) [inline lib/game.rb:107]: Nice and cleannn

**Pull request merged**